python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -534,3 +644,80 @@ class BaseRandomForestModel(Base, InteropMixin):
                 align_bytes=align_bytes,
             )
         return fil_model
+
+    @abstractmethod
+    def _compute_oob_score_metric(self, y_true, oob_predictions, valid_mask):
+        """
+        Compute the OOB score metric for the specific model type.
+
+        Parameters
+        ----------
+        y_true : array-like
+            True labels
+        oob_predictions : array-like
+            OOB predictions (1D for regression, 2D for classification)
+        valid_mask : array-like
+            Boolean mask indicating which samples have OOB predictions
+
+        Returns
+        -------
+        float
+            The computed score
+        """
+        pass
+
+    def _compute_oob_score(self, X, y):
+        """
+        Compute OOB score using per-tree predictions and bootstrap masks.
+        """
+        # Get per-tree predictions using FIL
+        fil_model = self.as_fil()
+        per_tree_preds = fil_model.predict_per_tree(X)
+        # Shape: (n_samples, n_trees) for regression
+        #        (n_samples, n_trees, n_classes) for classification
+
+        n_samples = X.shape[0]
+
+        # Determine output shape based on prediction dimensionality
+        # For regression: (n_samples,)
+        # For classification: (n_samples, n_classes)
+        output_shape = (n_samples,) + per_tree_preds.shape[2:]
+        oob_predictions = cp.zeros(output_shape, dtype=cp.float64)
+        oob_counts = cp.zeros(n_samples, dtype=cp.int32)
+
+        # For each tree, accumulate predictions for OOB samples
+        for tree_idx in range(self.n_estimators):
+            # Get OOB mask for this tree (samples NOT in bootstrap)
+            in_bag_mask = self._bootstrap_masks_[tree_idx]
+            oob_mask = ~in_bag_mask
+
+            # Accumulate predictions for OOB samples
+            oob_predictions[oob_mask] += per_tree_preds[oob_mask, tree_idx]
+            oob_counts[oob_mask] += 1
+
+        # Average OOB predictions (broadcasting handles both 1D and 2D cases)
+        valid_oob = oob_counts > 0
+
+        # Warn if some samples don't have OOB predictions
+        if not valid_oob.all():
+            warnings.warn(
+                "Some inputs do not have OOB scores. This probably means "
+                "too few trees were used to compute any reliable OOB estimates.",
+                UserWarning
+            )
+
+        if oob_predictions.ndim > 1:
+            oob_predictions[valid_oob] /= oob_counts[valid_oob, cp.newaxis]
+        else:
+            oob_predictions[valid_oob] /= oob_counts[valid_oob]
+
+        # Assign OOB predictions to the appropriate attribute based on estimator type
+        if self._estimator_type == "regressor":
+            self.oob_prediction_ = oob_predictions
+        else:
+            self.oob_decision_function_ = oob_predictions
```

Comment:
Per our existing reflection policies:

- `oob_prediction_` and `oob_decision_function_` should be `CumlArrayDescriptor`s defined on the respective classes.
- You might need to wrap these values into `CumlArray` before assigning them to get things to work properly. This is due to a bug in the reflection machinery that only shows up if internal code later references these attributes. If they're only user facing then nvm, everything will just work.

Additionally, can you add `oob_score_`/`oob_predition_`/`oob_decision_function_` to the docstring `Attributes` sections for their respective estimators?